Added option to reuse GitSCM's excluded users list

pbalint opened the pull request:
This change enables the github plugin to respect GitSCM's excluded user list (which is used when polling) for github webhooks:

![image](https://user-images.githubusercontent.com/4727754/68998493-3172d980-08b3-11ea-8088-c0e60863498d.png)

pbalint commented:
@KostyaSha , @lanwen May I get a review, please?

KostyaSha commented on src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java:
```diff
@@ -98,6 +104,22 @@ public void onPost(final GitHubTriggerEvent event) {
+                }
+            }
+
+            if (excludedUsers != null && excludedUsers.contains(event.getTriggeredByUser())) {
```
Should be ignorecase

pbalint replied:
It appears to me the polling is case sensitive as well:
https://github.com/jenkinsci/git-plugin/blob/a7ad6eb38a00397c4a656b53d99bfbad8ea3b2cc/src/main/java/hudson/plugins/git/extensions/impl/UserExclusion.java#L63
So if the webhooks checks the exclusion list in a case insensitive manner, there will be a counter-intuitive difference...

KostyaSha replied:
gitscm extension is for git users, github users case insensitive you do comparison in terms of GH functionality it should be ignorecase

pbalint replied:
Done

pbalint commented:
Ok, I've updated the PR and changed it to be case insensitive...

On Tue, Dec 17, 2019 at 7:26 PM Kanstantsin Shautsou <
notifications@github.com> wrote:

> *@KostyaSha* commented on this pull request.
> ------------------------------
>
> In src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java
> <https://github.com/jenkinsci/github-plugin/pull/222#discussion_r358955220>
> :
>
> > @@ -98,6 +104,22 @@ public void onPost(final GitHubTriggerEvent event) {
>          if (Objects.isNull(job)) {
>              return; // nothing to do
>          }
> +        if (useGitExcludedUsers) {
> +            Set<String> excludedUsers = null;
> +            if (job instanceof AbstractProject) {
> +                SCM scm = ((AbstractProject<?, ?>) job).getScm();
> +                if (scm instanceof GitSCM) {
> +                    UserExclusion exclusions = ((GitSCM) scm).getExtensions().get(UserExclusion.class);
> +                    if (exclusions != null) {
> +                        excludedUsers = exclusions.getExcludedUsersNormalized();
> +                    }
> +                }
> +            }
> +
> +            if (excludedUsers != null && excludedUsers.contains(event.getTriggeredByUser())) {
>
> gitscm extension is for git users, github users case insensitive you do
> comparison in terms of GH functionality it should be ignorecase
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jenkinsci/github-plugin/pull/222?email_source=notifications&email_token=ABECHSXCRWRN74ZDF7RJJRLQZEKW7A5CNFSM4JOGT3JKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCPQJQ4Y#discussion_r358955220>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABECHSS4477PWB2VJMLJMG3QZEKW7ANCNFSM4JOGT3JA>
> .
>
